src/extension/android/androidEmulatorManager.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import { AdbHelper } from "./adb";
+import { ChildProcess } from "../../common/node/childProcess";
+import { IEmulator, EmulatorManager } from "../EmulatorManager";
+import { OutputChannelLogger } from "../log/OutputChannelLogger";
+import * as nls from "vscode-nls";
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize = nls.loadMessageBundle();
+export interface IAndroidEmulator extends IEmulator {
+}
+
+export class AndroidEmulatorManager extends EmulatorManager{
+    private static readonly EMULATOR_COMMAND = "emulator";
+    private static readonly EMULATOR_LIST_AVDS_COMMAND = `-list-avds`;
+    private static readonly EMULATOR_AVD_START_COMMAND = `-avd`;
+
+    private static readonly EMULATOR_START_TIMEOUT = 120;
+
+    private logger: OutputChannelLogger = OutputChannelLogger.getChannel(OutputChannelLogger.MAIN_CHANNEL_NAME, true);
+
+    private adbHelper: AdbHelper;
+    private childProcess: ChildProcess;
+
+    constructor(adbHelper: AdbHelper) {
+        super();
+        this.adbHelper = adbHelper;
+        this.childProcess = new ChildProcess();
+    }
+
+    public async startEmulator(target: string): Promise<IAndroidEmulator | null> {
+        const onlineDevices = await this.adbHelper.getOnlineDevices();
+        for (let i = 0; i < onlineDevices.length; i++){
+            if (onlineDevices[i].id === target) {
+                return {id: onlineDevices[i].id};
+            }
+        }
+        if (target && (await this.adbHelper.getOnlineDevices()).length === 0) {
+            if (target === "simulator") {
```

Comment:
I think it is better to standartize such string literals, for example we can replace them by enums. Maybe not in this PR, but lets keep that in mind.